thread_sync.c

```diff
@@ -803,6 +803,8 @@ queue_do_push(VALUE self, struct rb_queue *q, VALUE obj)
  *
  * - when +empty?+ is false, calling deq/pop/shift will return an object
  *   from the queue as usual.
+ * - when +empty?+ is true, deq(non_block=false) will not suspend and will
```

Comment:
`deq(non_block=false)` includes unneeded local variable assignment.
How about mention both `deq(false)` and `deq(true)`?

Reply:
I updated the text based on your comments.